ledger/accountdb.go

```diff
@@ -1214,21 +1495,100 @@ func (ba *baseAccountData) GetAccountData() basics.AccountData {
 		MicroAlgos:         ba.MicroAlgos,
 		RewardsBase:        ba.RewardsBase,
 		RewardedMicroAlgos: ba.RewardedMicroAlgos,
-		VoteID:             ba.VoteID,
-		SelectionID:        ba.SelectionID,
-		StateProofID:       ba.StateProofID,
-		VoteFirstValid:     ba.VoteFirstValid,
-		VoteLastValid:      ba.VoteLastValid,
-		VoteKeyDilution:    ba.VoteKeyDilution,
 		AuthAddr:           ba.AuthAddr,
 		TotalAppSchema: basics.StateSchema{
 			NumUint:      ba.TotalAppSchemaNumUint,
 			NumByteSlice: ba.TotalAppSchemaNumByteSlice,
 		},
 		TotalExtraAppPages: ba.TotalExtraAppPages,
+
+		VoteID:          ba.VoteID,
+		SelectionID:     ba.SelectionID,
+		StateProofID:    ba.StateProofID,
+		VoteFirstValid:  ba.VoteFirstValid,
+		VoteLastValid:   ba.VoteLastValid,
+		VoteKeyDilution: ba.VoteKeyDilution,
+	}
+}
+
+// IsEmpty return true if any of the fields other then the UpdateRound are non-zero.
+func (bv baseVotingData) IsEmpty() bool {
+	return bv.VoteID.MsgIsZero() &&
+		bv.SelectionID.MsgIsZero() &&
+		bv.StateProofID.MsgIsZero() &&
+		bv.VoteFirstValid == 0 &&
+		bv.VoteLastValid == 0 &&
+		bv.VoteKeyDilution == 0
```

Comment:
nit: maybe something like this to avoid declaring everything twice, and risking that we forget to add it here if another field is needed later.
```suggestion
	zero := baseVotingData{UpdateRound: bv.UpdateRound}
	return zero == bv
```

Reply:
`return bv == baseVotingData{}` is even ~30% faster. Fixed.
but for `baseAccountData` this version `return ba == baseAccountData{UpdateRound: ba.UpdateRound}` is 2.5 times slower. Retained as is.